### ISSUE TITLE
Add shadow to keyboard keys

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#28]
### Terms

- [X] I have searched [open and closed design issues](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Adesign)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Once #24 is finished, the next step will be to add shadows to all keyboard keys. These changes will likely happen in [MyKeyboardView.kt](https://github.com/scribe-org/Scribe-Android/blob/main/app/src/main/kotlin/org/scribe/views/MyKeyboardView.kt).

### Contribution

This will become a priority after #24 is finished, and I'd be happy to work on it with people :)

**Comment by angrezichatterbox** [comment]
Hello There , I am new to open source and recently  attended a Wikimedia hackathon and learnt about contributing to repos related to Wikimedia  I would love to start contributing to a repository . I would start working on this issue

**Comment by andrewtavis** [comment]
Hey @angrezichatterbox 👋 Would be great to get some support here! As a community we're quite focused on Scribe-iOS and Scribe-Data right now, but if you do have interest in working on this I'd be happy to review the changes 😊

**Comment by angrezichatterbox** [comment]
I would like to confirm before I create a pull request.By making a shadow I have added a border around each key and also I have added a little color a shade of white . Is this what is required ??

**Comment by andrewtavis** [comment]
Hey @angrezichatterbox 👋 Is it a border or a shadow? We need it to be "beneath" the key and not just around the key :) As far as color, you can pick a middle grey value so it works on light and dark mode. I can edit it from there!

**Comment by angrezichatterbox** [comment]
I have created a pull request  Please do check it and tell me if any changes are required.

**Comment by angrezichatterbox** [comment]
Is it a good practice to work on the other issues while this is getting resolved ??

**Comment by andrewtavis** [comment]
You certainly could, @angrezichatterbox :) I'm in the midst of releasing both Scribe-Data and Scribe-iOS right now, and will have some more time for this once that's done. Apologies that your first experience is taking so long! These releases have been a long time coming :)

**Comment by andrewtavis** [comment]
I'd estimate that a check for the PR won't happen before early next week, just FYI. Again my apologies, but the other priorities are quite stacked up. If you would like to work on Scribe-Android a lot though, then I'd be happy to hop on a call at some point and we could map out some more issues. There's a LOT of interest in this project, including loads of Wikimedia Deutschland employees and members of the Wikidata community, but we just have a lot of things going on with the currently released applications.

**Comment by andrewtavis** [comment]
Plan is that some community members will switch over here in say the fall once Google Summer of Code for Scribe-iOS and Scribe-Data are done ☀️

**Comment by angrezichatterbox** [comment]
I would continue working on the scribe-android and work on the other issues in the meanwhile . I would love to work on this project and work on any issue that arises in the future as well.

**Comment by angrezichatterbox** [comment]
Thank you for giving me your time during your other projects

**Comment by andrewtavis** [comment]
More than happy to communicate on this, @angrezichatterbox! This is important work, and we will be focussing here eventually :) You'd also be welcome to contribute to Scribe-iOS if that'd be of interest :)

**Comment by andrewtavis** [comment]
Hey @angrezichatterbox 👋 Re #35, we now have the borders, which closes #24. It'd be great to also have shadows beneath the keys and make them white. So we don't actually need the border as described in #24 as that's an old design, but it'd be perfect if the same shapes that you have for the keys actually had small shadows under them, and that each of the keys was white. Would you have interest in working on that? I can explain more if it'd help! 😊

**Comment by angrezichatterbox** [comment]
I would love to work on it . Could u describe it a little more so that I would be able to do the right job on it

**Comment by andrewtavis** [comment]
Yes sure, @angrezichatterbox. Basically we're trying to get the interface to be very similar to Scribe-iOS'. For any key we should have:

- A white background (we can color special keys later)
- A small shadow under it
- No border around it

The shapes of the keys are great so far! Let me know if further explanation is needed 😊

**Comment by angrezichatterbox** [comment]
<img width="447" alt="Screenshot 2024-06-30 at 8 16 51 PM" src="https://github.com/scribe-org/Scribe-Android/assets/62511283/8819a2c7-79e9-4d46-a50b-c05181c833c7">

This is what I have made so far . What more changes should I make

**Comment by andrewtavis** [comment]
So far so good, @angrezichatterbox! From here we should remove the borders and add a small shadow under each of the keys. Then we'll be good for a PR!

**Comment by andrewtavis** [comment]
Further notes on this, @angrezichatterbox: If we can reduce the corner radius by say 1/3 it's current value, and also there's a bit too much horizontal space between the keys. We can then move on to another issue :)

**Comment by angrezichatterbox** [comment]
[
<img width="447" alt="Screenshot 2024-07-01 at 2 17 51 PM" src="https://github.com/scribe-org/Scribe-Android/assets/62511283/18920a78-e25a-4ab0-ae2f-3e6b965699b4">
](url)

This is what I have made . Is this Good enough or Should I make any more changes

**Comment by andrewtavis** [comment]
This is 100% ok, @angrezichatterbox :) We're good for a PR 😊

**Comment by angrezichatterbox** [comment]
I have made the PR . Please do say if u have any suggestions on making my commit messages and branch naming style or Pull request naming better

**Comment by andrewtavis** [comment]
Looks good so far, @angrezichatterbox! I'd always use present tense in branch names, so `make-keyboard` instead of `making-keyboard`. Next time :) How about I add a PR template to the repo that gives you some directions on your commit message as well?

**Comment by angrezichatterbox** [comment]
Thanks for your suggestion , I would try to follow those while solving the next issue . It would be nice if there would be a template that gives direction on PR and commit message .

**Comment by andrewtavis** [comment]
I send that along tonight then after I review the PR 😊

**Comment by andrewtavis** [comment]
Btw, @angrezichatterbox, it would be great to have you in the [public Matrix chat rooms](https://matrix.to/#/%23scribe_community:matrix.org) :) From there we can figure out a time for a call, or also feel free to email me at the email in my profile 😊

**Comment by andrewtavis** [comment]
Closed via #37 :) Slowly coming along, @angrezichatterbox! Great to have all of this progress 😊
